refinery/core/models.py

```diff
@@ -613,15 +607,17 @@ def get_latest_investigation_link(self, version=None):
         """
         try:
             if version is None:
-                version = InvestigationLink.objects.filter(
-                        data_set=self
-                    ).aggregate(Max("version"))["version__max"]
+                return InvestigationLink.objects.filter(
+                    data_set=self
+                ).latest('date')
 
             return InvestigationLink.objects.get(
                 data_set=self,
                 version=version
             )
-        except:
+        except (InvestigationLink.DoesNotExist,
+                InvestigationLink.MultipleObjectsReturned) as exc:
+            logger.error("Couldn't properly fetch InvestigationLink: %s", exc)
```

Comment:
I think it can happen pretty easily. I believe there is a window during metadata upload where a `DataSet` has been created, but its corresponding `InvestigationLink` hasn't been.

Reply:
So no need to log the error in that case?

Reply:
Probably not
Maybe the `MultipleObjectsReturned` if anything since that would be really odd